nucliadb_core/src/metrics/meters/prometheus.rs

```diff
@@ -69,20 +82,46 @@ impl Meter for PrometheusMeter {
 }
 
 impl PrometheusMeter {
-    pub fn new() -> PrometheusMeter {
+    pub fn new() -> Self {
         let mut registry = Registry::default();
 
         // This must be done for every metric
         let request_time_metric = request_time::register_request_time(&mut registry);
+        let tokio_runtime_metrics = tokio_runtime::register_tokio_runtime_metrics(&mut registry);
         let tokio_task_metrics = tokio_tasks::register_tokio_task_metrics(&mut registry);
 
         let tasks_monitor = MultiTaskMonitor::new();
+        let runtime_monitor = RuntimeMonitor::new(&tokio::runtime::Handle::current());
+        let runtime_intervals = Mutex::new(runtime_monitor.intervals());
 
-        PrometheusMeter {
+        Self {
             registry,
             request_time_metric,
+            tokio_runtime_metrics,
             tokio_task_metrics,
             tasks_monitor,
+            runtime_intervals,
+        }
+    }
+
+    fn collect_runtime_metrics(&self) -> NodeResult<()> {
+        match self.runtime_intervals.try_lock() {
+            Ok(ref mut intervals) => {
+                if let Some(metrics) = intervals.next() {
+                    self.tokio_runtime_metrics.collect(metrics);
+                } else {
+                    return Err(node_error!(
+                        "Cannot export tokio runtime metrics, iterator didn't return values"
+                    ));
+                }
+            }
+            Err(error) => {
+                return Err(node_error!(
+                    "Cannot acquire runtime metrics lock. There's a concurrent export going on? \
+                     {error}"
+                ));
+            }
```

Comment:
A `try_lock` can also return a `PoisonError`, which I think should be treated differently, in this case I would just get the inner lock using `into_inner` and collect either way.
[More info](https://doc.rust-lang.org/std/sync/enum.TryLockError.html) 
```rust
match self.runtime_intervals.try_lock() {
       Ok(..) => {..}
       Err(TryLockError::Poisoned(inner)) => { I would get the inner lock and collect }
       Err(TryLockError::WouldBlock) => { return the error if you want} 
}

Reply:
Solved